pkg/tekton/metapipeline/metapipeline.go

```diff
@@ -0,0 +1,233 @@
+
+	tektonCRDs, err := tekton.NewCRDWrapper(pipeline, tasks, resources, structure, run)
+	if err != nil {
+		return nil, err
+	}
+
+	return tektonCRDs, nil
+}
+
+// createPipeline builds the parsed/typed pipeline which servers as input for the Tekton CRD creation.
+func createPipeline(params CRDCreationParameters) (*syntax.ParsedPipeline, error) {
+	steps, err := BuildExtensionSteps(params.Apps)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to create app extending pipeline steps")
+	}
+
+	stage := syntax.Stage{
+		Name:  appExtensionStageName,
+		Steps: steps,
+		Agent: &syntax.Agent{
+			// TODO Agent does not seem to be used anywhere
+			Image: "unused",
+		},
+	}
+
+	parsedPipeline := &syntax.ParsedPipeline{
+		Stages: []syntax.Stage{stage},
+	}
+
+	env := buildEnvParams(params)
+	parsedPipeline.AddContainerEnvVarsToPipeline(env)
+
+	return parsedPipeline, nil
+}
+
+// GetExtendingApps returns the lst of apps which are installed in the cluster registered for extending the pipeline.
+// An app registers its interest in extending the pipeline by having the 'pipeline-extension' label set.
+func GetExtendingApps(jxClient versioned.Interface, namespace string) ([]jenkinsv1.App, error) {
+	listOptions := metav1.ListOptions{}
+	listOptions.LabelSelector = fmt.Sprintf(apps.AppTypeLabel+" in (%s)", apps.PipelineExtension)
+	appsList, err := jxClient.JenkinsV1().Apps(namespace).List(listOptions)
+	if err != nil {
+		return nil, errors.Wrap(err, "error  retrieving pipeline contributor apps")
+	}
+	return appsList.Items, nil
+}
+
+// BuildExtensionSteps builds a step (container) for each of the apps specified.
+func BuildExtensionSteps(extendingApps []jenkinsv1.App) ([]syntax.Step, error) {
+	var steps []syntax.Step
+
+	// TODO - Add step to create effective pipeline. This way apps will always have a pipeline config to load and work with,
+	// TODO - even if the source uses an implicit pipeline config
+	// TODO - In case of a release we might want to add a task to create the tag
+
+	log.Debugf("creating pipeline steps for extending apps")
+	for _, app := range extendingApps {
+		if app.Spec.PipelineExtension == nil {
+			log.Warnf("Skipping app %s in meta pipeline. It contains label %s with value %s, but does not contain PipelineExtension fields.", app.Name, apps.AppTypeLabel, apps.PipelineExtension)
+			continue
+		}
+
+		extension := app.Spec.PipelineExtension
+		step := syntax.Step{
+			Name:      extension.Name,
+			Image:     extension.Image,
+			Command:   extension.Command,
+			Arguments: extension.Args,
+		}
+
+		log.Debugf("App %s contributes with step %s", app.Name, util.PrettyPrint(step))
+		steps = append(steps, step)
+	}
+
+	// TODO - Add step which does call the 'step create task' functionality to build the actual build pipeline
+	return steps, nil
+}
+
+func buildEnvParams(params CRDCreationParameters) []corev1.EnvVar {
+	var envVars []corev1.EnvVar
+
+	envVars = append(envVars, corev1.EnvVar{
```

Comment:
I might be missing some variables, but the ones here make obvious sense right now. If we need more and in particular if more is needed for creating the build pipeline needs to be seen.